File: Go/linearsearch.go
// Linear Search in Go

package main

import "fmt"

func main() {

	var arr = []int{2, 3, 4, 10, 40}
	var x = 10
	fmt.Println(linearSearch(arr, x))

}

func linearSearch(arr []int, x int) int {

	for i := 0; i < len(arr); i++ {
		if arr[i] == x {
			return i
		}

	}
	return -1
}

// Output:
// 3

File: Rust/linearsearch.rs
// Linear Search in Rust
use std::convert::TryInto;
fn main() {
    let arr = [2, 3, 4, 10, 40];
    let x = 10;
    println!("{}", linear_search(arr, x));
}

fn linear_search(arr: [i32; 5], x: i32) -> i32 {
    let mut i = 0;
    while i < arr.len() {
        if arr[i] == x {
            return i.try_into().unwrap();
        }
        i += 1;
    }
    return -1;
}

// Output:
// 3
